R/CohortRelationship.R

```diff
@@ -0,0 +1,538 @@
+              "CreatedDiagnosticsCohortRelationship.csv"
+            ),
+            force = TRUE
+          )
+        } else {
+          timePeriodsPreviouslyExecuted <- previousRunResults %>%
+            dplyr::select("startDay",
+                          "endDay",
+                          "timeId") %>%
+            dplyr::distinct()
+          timePeriodsPreviouslyExecuted <-
+            timePeriodsPreviouslyExecuted %>%
+            dplyr::inner_join(timePeriods,
+                              by = c("startDay",
+                                     "endDay",
+                                     "timeId")) %>%
+            dplyr::select(.data$timeId) %>%
+            dplyr::distinct()
+          
+          ParallelLogger::logTrace(
+            paste0(
+              "    - Found previous execution to have ",
+              scales::comma(nrow(timePeriodsPreviouslyExecuted)),
+              " records. Removing time_periods corresponding to those executions and reusing previous results."
+            )
+          )
+          timePeriods <- timePeriods %>%
+            dplyr::anti_join(timePeriodsPreviouslyExecuted,
+                             by = c("timeId"))
+          
+          ParallelLogger::logTrace(paste0(
+            "    - Executing over ",
+            scales::comma(nrow(timePeriods)),
+            " time_periods."
+          ))
+          
+          resultsInAndromeda$cohortRelationships <-
+            previousRunResults %>%
+            dplyr::select(-.data$startDay,-.data$endDay)
+        }
+      } else {
+        ParallelLogger::logTrace("    - Not found, running cohort relationship diagnostics for all time periods.")
+      }
+    }
+    for (i in (1:nrow(timePeriods))) {
+      ParallelLogger::logTrace(
+        paste0(
+          "    - Working on ",
+          scales::comma(timePeriods[i,]$startDay),
+          " to ",
+          scales::comma(timePeriods[i,]$endDay),
+          " days (",
+          scales::comma(i),
+          " of ",
+          scales::comma(nrow(timePeriods)),
+          ")"
+        )
+      )
+      sql <- SqlRender::loadRenderTranslateSql(
+        "CohortRelationship.sql",
+        packageName = "CohortDiagnostics",
+        dbms = connection@dbms,
+        time_id = timePeriods[i, ]$timeId,
+        start_day_offset = timePeriods[i, ]$startDay,
+        end_day_offset = timePeriods[i, ]$endDay
+      )
+      DatabaseConnector::querySqlToAndromeda(
+        connection = connection,
+        sql = sql,
+        snakeCaseToCamelCase = TRUE,
+        andromeda = resultsInAndromeda,
+        andromedaTableName = 'temp'
+      )
+      
+      if (!"cohortRelationships" %in% names(resultsInAndromeda)) {
+        resultsInAndromeda$cohortRelationships <- resultsInAndromeda$temp
+      } else {
+        Andromeda::appendToTable(resultsInAndromeda$cohortRelationships,
+                                 resultsInAndromeda$temp)
+      }
+      readr::write_excel_csv(
```

Comment:
this is what is causing github actions to fail. Test code on local first. This is easy with devtools::test() which only runs the sqlite tests for this package so is very quick. You can also run just the tests for this PR with 

```
devtools::test(filter="moduleCohortRelationship")
```

Reply:
thats a good tip!